### ISSUE TITLE
Gtfs reader fails if agency is null

### ISSUE THREAD

**Issue by karussell** [graphhopper/graphhopper#1514]
The reason is that in `Route$Loader.loadOneRow` the map `feed.agency` is empty. See forum post: https://discuss.graphhopper.com/t/issues-setting-up-gtfs-routing/3978

When I run this the first time I get:

```
INFO  [2019-01-10 08:49:48,492] com.conveyal.gtfs.model.Entity$Loader: Table feed_info was missing but it is not required.
INFO  [2019-01-10 08:49:48,492] com.conveyal.gtfs.GTFSFeed: Feed ID is undefined, pester maintainers to include a feed ID. Using file name gtfs.
INFO  [2019-01-10 08:49:48,493] com.conveyal.gtfs.model.Entity$Loader: Loading GTFS table agency from agency.txt
INFO  [2019-01-10 08:49:48,496] com.conveyal.gtfs.model.Entity$Loader: Loading GTFS table calendar from calendar.txt
INFO  [2019-01-10 08:49:48,514] com.conveyal.gtfs.model.Entity$Loader: Loading GTFS table calendar_dates from calendar_dates.txt
INFO  [2019-01-10 08:49:48,602] com.conveyal.gtfs.model.Entity$Loader: Table fare_attributes was missing but it is not required.
INFO  [2019-01-10 08:49:48,603] com.conveyal.gtfs.model.Entity$Loader: Table fare_rules was missing but it is not required.
INFO  [2019-01-10 08:49:48,604] com.conveyal.gtfs.model.Entity$Loader: Loading GTFS table routes from routes.txt
java.lang.RuntimeException: java.lang.NullPointerException
	at com.graphhopper.reader.gtfs.GtfsStorage.loadGtfsFromFile(GtfsStorage.java:203)
	at com.graphhopper.reader.gtfs.GraphHopperGtfs.createOrLoad(GraphHopperGtfs.java:347)
	at com.graphhopper.http.GraphHopperBundle.runPtGraphHopper(GraphHopperBundle.java:213)
	at com.graphhopper.http.GraphHopperBundle.run(GraphHopperBundle.java:202)
	at com.graphhopper.http.GraphHopperBundle.run(GraphHopperBundle.java:64)
	at io.dropwizard.setup.Bootstrap.run(Bootstrap.java:200)
	at io.dropwizard.cli.EnvironmentCommand.run(EnvironmentCommand.java:42)
	at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:87)
	at io.dropwizard.cli.Cli.run(Cli.java:78)
	at io.dropwizard.Application.run(Application.java:93)
	at com.graphhopper.http.GraphHopperApplication.main(GraphHopperApplication.java:33)
Caused by: java.lang.NullPointerException
	at com.conveyal.gtfs.model.Route$Loader.loadOneRow(Route.java:83)
	at com.conveyal.gtfs.model.Entity$Loader.loadTable(Entity.java:269)
	at com.conveyal.gtfs.GTFSFeed.loadFromFile(GTFSFeed.java:163)
	at com.conveyal.gtfs.GTFSFeed.loadFromFile(GTFSFeed.java:174)
	at com.graphhopper.reader.gtfs.GtfsStorage.loadGtfsFromFile(GtfsStorage.java:199)
	... 10 more
```

If we cannot properly handle this we should throw a better error message and what might be wrong with the data :)

And when I execute this again I get #1513, which should not be.

**Comment by michaz** [comment]
I'll fix #1513 (that one is really bad), but I'm not sure if I want to start doing in-depth validation on GTFS files at this point. Right now, I think _any_ referential inconsistency will come out as a ``NullPointerException`` in the end.

**Comment by michaz** [comment]
(In this case, a required line in ``agency.txt`` is missing.)

**Comment by michaz** [comment]
(The parts of GTFSLib we're still using already does some of the work and marks most things it thinks suspicious. (In this specific case, that fails.) But we're currently ignoring those marks. I could check if there are "no errors" before accepting a GTFS file, but maybe we would be too careful then an reject GTFS files which would work?)

**Comment by michaz** [comment]
You know what, let's just put a message in the top-level ``catch`` block that says:

``Error while building the GTFS model. Please run your GTFS file through some validator to check if your GTFS file is valid. If it is, please raise an issue with us.``

**Comment by karussell** [comment]
(fixed via #1523 - thanks!)
